
Written for Archivist, allow printing all unified as the Anonymous user.

  - Rene Rebe <rene@exactcode.de>

diff -ur cups-pdf-2.0beta2/extra/cups-pdf.conf cups-pdf-2.0beta2-tweaked/extra/cups-pdf.conf
--- cups-pdf-2.0beta2/extra/cups-pdf.conf	2005-10-26 12:15:51.000000000 +0000
+++ cups-pdf-2.0beta2-tweaked/extra/cups-pdf.conf	2005-12-03 13:13:01.000000000 +0000
@@ -57,6 +57,14 @@
 
 #AnonDirName /var/spool/cups-pdf/ANONYMOUS
 
+### Key: AllAnonUser
+##  Print any job as anonymous user. Useful for unified server
+##  side postprocessing, such as injecting into a database.
+### Default: 0
+
+#AllAnonUser 0
+
+
 ### Key: Spool
 ##  CUPS-PDF spool directory - make sure there is no user 'SPOOL' on your
 ##  system or change the path	
diff -ur cups-pdf-2.0beta2/src/cups-pdf.c cups-pdf-2.0beta2-tweaked/src/cups-pdf.c
--- cups-pdf-2.0beta2/src/cups-pdf.c	2005-10-23 14:04:45.000000000 +0000
+++ cups-pdf-2.0beta2-tweaked/src/cups-pdf.c	2005-12-03 13:08:36.000000000 +0000
@@ -393,7 +393,8 @@
     snprintf(user, size, "%s%s", conf.userprefix, argv[2]);
     passwd=getpwnam(user);
   }  
-  if (passwd == NULL) {
+  if (passwd == NULL || conf.allanonuser == 1) {
+	  passwd = NULL;
     if (strlen(conf.anonuser)) {
       passwd=getpwnam(conf.anonuser);
       if (passwd == NULL) {
diff -ur cups-pdf-2.0beta2/src/cups-pdf.h cups-pdf-2.0beta2-tweaked/src/cups-pdf.h
--- cups-pdf-2.0beta2/src/cups-pdf.h	2005-10-23 14:03:20.000000000 +0000
+++ cups-pdf-2.0beta2-tweaked/src/cups-pdf.h	2005-12-03 13:10:40.000000000 +0000
@@ -56,6 +56,7 @@
   short		label;
   short 	logtype;
   short		lowercase;
+	short		allanonuser;
   mode_t	anonumask;
   mode_t	userumask;
 } conf;
@@ -79,6 +80,7 @@
   conf.label=0;
   conf.logtype=3;
   conf.lowercase=1;
+	conf.allanonuser=0;
   conf.anonumask=0000;
   conf.userumask=0077; 
   return;
@@ -134,6 +136,10 @@
     tmp=atoi(value);
     conf.lowercase=(tmp>0)?1:0;
   }
+  else if (!strcmp("AllAnonUser",key)) {
+    tmp=atoi(value);
+	  conf.allanonuser=(tmp>0)?1:0;
+  }
   else if (!strcmp("AnonUMask",key)) {
     tmp=atoi(value);
     conf.anonumask=tmp;
